packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx

```diff
@@ -0,0 +1,45 @@
+import { createElement, ReactNode, ReactElement } from "react";
+import { ObjectItem } from "mendix";
+import Swiper, { SwiperOptions, Navigation, Pagination, EffectFade, Autoplay } from "swiper";
+import { Swiper as SwiperReact, SwiperSlide } from "swiper/react";
+
+export interface CarouselProps<T extends ObjectItem> {
+    pagination: boolean;
+    loop: boolean;
+    tabIndex?: number | undefined;
+    animation?: boolean;
+    autoplay?: boolean;
+    delay?: number;
+    navigation: boolean;
+    onClick: () => void;
+    items: T[];
```

Comment:
Isn't T just `ObjectItem` here? So instead of typing it as `T[]`, `ObjectItem[]` would also work + itemRenderer second argument woud also be `ObjectItem`. Is there a particular benefit to use the generic typing instead of `ObjectItem`?